packages/hardhat-ledger/src/internal/with-spinners.ts

```diff
@@ -35,9 +39,15 @@ function attachSpinner(
   const { startText, eventPrefix } = spinnerOptions;
   const spinner = ora({ text: startText, discardStdin: false });
 
-  emmiter.on(`${eventPrefix}_start`, () => spinner.start());
-  emmiter.on(`${eventPrefix}_success`, () => spinner.succeed());
-  emmiter.on(`${eventPrefix}_failure`, () => spinner.fail());
+  emitter.on(`${eventPrefix}_start`, () =>
+    emitter.isOutputEnabled ? spinner.start() : undefined
```

Comment:
No need to change.

Can I check. The use of the conditional operator here is equivalent to:

```ts
if (!emitter.isOutputEnabled) {
  return
}

spinner.start();
```

This is a style question, and not a blocker (nor even necessarily right 😉). When I read a conditional operator I assume it is returning a usable value. I also expect it to have no side-effects.

The functional programmer in me wants to be explicit about state updates and side-effects. So I typically go for obviously procedural `if` statements for side-effects (and for loops rather than maps/folds) and conditional operators and maps/folds for pure/transform code with no side effects.

This may be a convention I entirely made up, but I found the conditional operator here unexpected.

Reply:
Thanks for the insight. AFAIK, we don't have rules about ternary operators and side-effects. I see your point about being clear with state updates and effects. If it makes things clearer, I'm all for changing it.